openapi/pex-openapi.yaml

```diff
@@ -339,29 +339,62 @@ components:
       title: Descriptor
       description: descriptor map laying out the structure of the presentation submission.
       type: object
-      properties:
-        id:
-          type: string
-          description: ID to identify the descriptor from Presentation Definition Input Descriptor it coresponds to.
-        path:
-          type: string
-          description: The path where the verifiable credential is located in the presentation submission json
-        path_nested:
-          $ref: '#/components/schemas/descriptor'
-        format:
-          type: string
-          description: The Proof or JWT algorithm that the proof is in
-      required:
-        - id
-        - path
-        - format
-      additionalProperties: false
+      oneOf:
+        - type: object
+          title: StandardDescriptor
+          properties:
+            id:
+              type: string
+              description: ID to identify the descriptor from Presentation Definition Input Descriptor it coresponds to.
+            path:
+              type: string
+              description: The path where the verifiable credential is located in the presentation submission json
+            path_nested:
+              $ref: '#/components/schemas/descriptor'
+            format:
+              type: string
+              description: The Proof or JWT algorithm that the proof is in
+            cryptosuite:
```

Comment:
What is the need for the cryptosuite in this branch of the oneOf?

Reply:
Maybe there is a better way to solve this.
Ideally, I would like typescript to be able to narrow down the type of a descriptor to a DataIntegrityDescriptor 
if the format is equal to `DataIntegrityProof`.

```ts
export type Descriptor = DataIntegrityDescriptor | StandardDescriptor;
```

The problem is that the format field of the standard descriptor can be any string, i.e., it can also be 'DataIntegrityProof'; therefore, this does not work.

```ts
if (descriptor.format === 'DataIntegrityProof') {
    // Property cryptosuite does not exist on type ...
    descriptor.cryptosuite
}
```

That's why I added the cryptosuite in the other oneOf branch aswell.

Reply:
You can probably do this with an assertion function:

```ts
isDataIntegrityDescriptor(descriptor: Descriptor): descriptor is DataIntegrityDescriptor {
 return descriptor.format === 'DataIntegrityProof'
}
```


However, isn't format in descriptor used for `vc+sd-jwt`, `di_vc`, `jwt_vp`, etc.. instead of `DataIntegrityProof`? So format in this case would be `di_vc`, `di_vp` or `di`?

Reply:
Yeah agree, format should be `di_vc`, `di_vp` or `di` if you ask me. 

Regarding the narrowing down discussion. Although the spec/model currently mainly is used for Typescript its intention is that it could be used for/by other languages as well. We are using it internally for some Java/Kotlin as well for instance. I am always a bit hesitant to allow the spec to easy the implementation language.

I guess that @TimoGlastra solution, but then with the updated format values, would provide an elegant solution in case you need to access the `cryptosuite` value. 

Opinions?

Reply:
If we proceed as Timo has suggested below, we won't have this problem at all.